src/routes/package-show.js

```diff
@@ -173,21 +181,80 @@ class PackageShowRoute extends Component {
 
   componentWillUnmount() {
     this.props.clearCostPerUseData();
+    window.clearTimeout(this.timeout);
   }
 
   /* This method is common between package-show and package-edit routes
    * This should be refactored once we can share model between the routes.
   */
   addPackageToHoldings = () => {
-    const { model, updatePackage } = this.props;
+    const {
+      model,
+      match,
+      updatePackage,
+    } = this.props;
+    const { packageId } = match.params;
+
     model.isSelected = true;
     model.selectedCount = model.titleCount;
     model.allowKbToAddTitles = true;
+
     updatePackage(model);
+    this.updateTitles(packageId);
   };
 
+  getUpdatedTitles(packageId) {
+    const { getPackageTitles } = this.props;
+    const { pkgSearchParams } = this.state;
+
+    const params = transformQueryParams('titles', pkgSearchParams);
+
+    this.setState(() => {
+      return { isTitlesUpdating: true };
+    });
+
+    this.timeout = window.setTimeout(() => {
+      getPackageTitles({ packageId, params });
+
+      this.setState(() => {
+        return { isTitlesUpdating: false };
+      });
+    }, 6000);
```

Comment:
It will be better to have  a constant, but why it is 6 seconds?

Reply:
Because for the less time there are often cases when data in the database isn't updated yet and request returns old values.